src/NServiceBus.Metrics/ResetMetricTimer.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Features;
+
+class ResetMetricTimer
+{
+    public event EventHandler NoMessageSentForAWhile;
+
+    public ResetMetricTimer(FeatureConfigurationContext featureConfigurationContext)
+    {
+        timer = new System.Threading.Timer(ResetCounterValueIfNoMessageHasBeenProcessedRecently, null, 0, 2000);
+
+        featureConfigurationContext.Pipeline.OnReceivePipelineCompleted(e =>
+        {
+            DateTime timeSent;
+            if (e.TryGetTimeSent(out timeSent))
+            {
+                lastMessageProcessedTime = e.CompletedAt;
+            }
+
+            return Task.FromResult(0);
+        });
+
+    }
+
+    void ResetCounterValueIfNoMessageHasBeenProcessedRecently(object state)
+    {
+        // Concurreny : This runs every 2s and might just fire before an update
```

Comment:
It might have not raised any issues as performance counters do not disturb the logic of execution. 
It's not about missing one update. It's about correctness and missing some percentage of them (maybe all). Again, this value does not disturb any logic. I really dunno.